File: python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self._adjust(hour, minute)

    # Public method
    def add(self, minute):
        self._adjust(self.hour, self.minute + minute)
        return self

    # Private method
    def _adjust(self, hour, minute):
        total_minute = hour * 60 + minute

        self.minute = total_minute % 60
        self.hour = (total_minute // 60) % 24

    # Override method
    def __str__(self):
        return '%0.2d:%0.2d' % (self.hour, self.minute)

    def __eq__(self, other):
        return str(self) == str(other)

File: ruby/clock/clock.rb
class Clock
  def initialize(min)
    @min = min
  end

  def self.at(hour, min)
    total_min = hour * 60 + min
    new(total_min)
  end

  def to_s
    hour = (@min / 60) % 24
    min = @min % 60
    "#{hour.to_s.rjust(2, '0')}:#{min.to_s.rjust(2, '0')}"
  end

  def +(min_amount)
    self.class.new(@min + min_amount)
  end

  def ==(other)
    return false unless other.class == Clock
    to_s == other.to_s
  end
end

module BookKeeping
  VERSION = 2
end
